eng/common/TestResources/deploy-test-resources.yml

```diff
@@ -43,6 +43,18 @@ steps:
   - template: /eng/common/TestResources/setup-environments.yml
 
   - ${{ if eq('true', parameters.UseFederatedAuth) }}:
+    - task: AzureCLI@2
+      displayName: Set OIDC variables
+      inputs:
+        azureSubscription: ${{ parameters.ServiceConnection }}
+        scriptType: pscore
+        scriptLocation: inlineScript
+        addSpnToEnvironment: true
+        inlineScript: |
+          Write-Host "##vso[task.setvariable variable=ARM_CLIENT_ID;issecret=true]$($env:servicePrincipalId)"
+          Write-Host "##vso[task.setvariable variable=ARM_TENANT_ID;issecret=true]$($env:tenantId)"
+          Write-Host "##vso[task.setvariable variable=ARM_OIDC_TOKEN;issecret=true]$($env:idToken)"
```

Comment:
I was copying this off the task you added. Want me to edit both and normalize?

Reply:
Which task are you referring to? I don't remember using OIDC naming anywhere.

Reply:
@weshaggard https://github.com/Azure/azure-sdk-for-net/pull/43912

Reply:
It wasn't an issue with my manual tests running the identity pipeline, and we have a fairly complex arm deployment.

Reply:
Also, at least in Identity's case, we only need the token in the -pre script, so that is well before the long part of the arm deployment. So perhaps we should document a best practice for deployments that need this, they should `az login` in a pre script.

Reply:
Even if they az login, the token used for login will still expire right? But maybe similar to what Wes said in the other comment thread we can possibly renew the access token in the deploy script if it becomes a problem for people.

Reply:
No, the OIDC token is exchanged for a regular access token, and that token would typically be good for at least an hour.

Reply:
Ah, great then.